Applications/Spire/Source/TimeAndSales/TimeAndSalesWindowModel.cpp

```diff
@@ -106,18 +130,38 @@ QVariant TimeAndSalesWindowModel::headerData(int section,
       case Columns::PRICE_COLUMN:
         return tr("Price");
       case Columns::SIZE_COLUMN:
-        return tr("Qty");
+        if(is_short_text(section, BREAKPOINT(m_header_font, tr("Quantity")))) {
+          return tr("Qty");
+        }
+        return tr("Quantity");
       case Columns::MARKET_COLUMN:
-        return tr("Mkt");
+        if(is_short_text(section, BREAKPOINT(m_header_font, tr("Market")))) {
+          return tr("Mkt");
+        }
+        return tr("Market");
       case Columns::CONDITION_COLUMN:
-        return tr("Cond");
+        if(is_short_text(section, BREAKPOINT(m_header_font,
+            tr("Condition")))) {
+          return tr("Cond");
+        }
+        return tr("Condition");
       default:
         return QVariant();
     }
+  } else if(role == Qt::FontRole) {
+    return m_header_font;
   }
   return QVariant();
 }
 
+bool TimeAndSalesWindowModel::is_short_text(int column, int breakpoint) const {
+  if(m_column_size_reference.find(column) !=
```

Comment:
In cases like this, use one lookup instead:

```
auto it = m_column_size_reference.find(column);
if(it != m_column_size_reference.end()) {
  return it->second < breakpoint;
}
```

As an aside, the advantage of method `at` over `operator []` is that the former will throw when the element is not present, as opposed to default constructing (in case of maps) or undefined behavior (in case of lists). That said, when you have already checked for existence of the element e.g. using `find`, using `at` is excessive (in single-threaded environments at least). The presence of the exception handling in the compiled code imposes quite some overhead so, when the element is definitely present, you should prefer calling `operator []` instead.